server/src/main/java/org/elasticsearch/index/codec/TrackingPostingsInMemoryBytesCodec.java

```diff
@@ -104,10 +106,13 @@ public Terms terms(String field) throws IOException {
                 return null;
             }
             int fieldNum = fieldInfos.fieldInfo(field).number;
-            return new TrackingLengthTerms(
-                terms,
-                bytes -> termsBytesPerField.put(fieldNum, Math.max(termsBytesPerField.getOrDefault(fieldNum, 0), bytes))
-            );
+            if (seenFields.add(fieldNum)) {
+                return new TrackingLengthTerms(terms, bytes -> totalBytes[0] += bytes);
+            } else {
+                // As far as I know only when bloom filter for _id filter gets written this method gets invoked twice for the same field.
+                // So maybe we can get rid of the seenFields here? And just keep track of whether _id field has been seen?
+                return terms;
```

Comment:
I think it's difficult to guarantee that we'll only ever invoke this method twice for `_id` fields. I think only checking the `_id` field would potentially open us up to some subtle bugs in the future.